src/flatc_main.cpp

```diff
@@ -216,42 +145,75 @@ int main(int argc, const char *argv[]) {
   std::shared_ptr<flatbuffers::CodeGenerator> ts_generator =
       flatbuffers::NewTsCodeGenerator();
 
-  flatc.RegisterCodeGenerator("--binary", binary_generator);
-  flatc.RegisterCodeGenerator("-b", binary_generator);
+  flatc.RegisterCodeGenerator(
+      flatbuffers::FlatCOption{
+          "b", "binary", "",
+          "Generate wire format binaries for any data definitions" },
+      binary_generator);
```

Comment:
since we aren't sharing the generators now, maybe just inline the `flatbuffers::New{LANG}Generator()` here.

Ditto for all the below generators.